src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Repository/ServiceRepository.php

```diff
@@ -131,4 +132,16 @@ public function findByTeamName(?string $serviceTeamName): ?Service
             ]
         );
     }
+
+    public function getByLastInsertId(): ?Service
+    {
+        $em = $this->getEntityManager();
+        try {
+            $lastId = (int) $em->getConnection()->lastInsertId();
```

Comment:
I think the most reliable way given cqrs is used.
This returns the last inserted pk of entities into the service table, in the **current connection**. So no issues with race conditions *
I since learned that the teamname is also unique, but this uniqueness is not enforced at database level, so I didn't trust it.

I added a check to be safe:
```
        if ($command->getTeamName() !== $service->getTeamName()) {
            throw new InviteException('Could not create invite due to a concurrency issue.');
        }
```

I agree it's not very elegant..

---

*Unless we use FrankenPHP to host the app in a mode that keeps symfony bootstrapped in the workers and thus may reuse the connection? (I don't think that will break it either, since each worker would still have it's own connection. I don't think the connection can be shared over workers.)

Reply:
@MKodde Thanks for the discussion and the tip to break cqrs more elegantly.
I updated the PR